bindings/cs/MongoDB.Crypt/CryptOptions.cs

```diff
@@ -14,28 +14,47 @@
  * limitations under the License.
  */
 
+using System.Collections.Generic;
+using System.Linq;
 
 namespace MongoDB.Crypt
 {
     /// <summary>
-    /// Options to configure mongocrypt with
+    /// Options to configure mongocrypt with.
     /// </summary>
     public class CryptOptions
     {
-        public IKmsCredentials KmsCredentials { get; }
+        public Dictionary<KmsType, IKmsCredentials> KmsCredentialsOptions { get; } = new Dictionary<KmsType, IKmsCredentials>();
         public byte[] Schema { get; }
 
-        public CryptOptions(IKmsCredentials kmsCredentials)
+        public CryptOptions(IEnumerable<IKmsCredentials> kmsCredentialsOptions) : this(kmsCredentialsOptions, null)
```

Comment:
Since it's invalid to have more than one type of kmsCredential,  how do you feel about the structuring our constructors to reflect this? One constructor for local credentials, one constructor for aws credentials, and one with both.

Reply:
I thought about this solution, my idea was to avoid straight references on `IKmsCredentials` implementations. It will help us to leave this class without any changes in case if we need to add some other implementation of `IKmsCredentials`, but maybe it's not a big advantage, since in any case, we will need to make changes in this(wrapper) solution (because I don't think that we will add a new possible `IKmsCredentials` implementation in `driver` itself)